tests/integration/it_roles_test.py

```diff
@@ -17,8 +18,12 @@ def test_create(client, db_session):
     assert Role.query.filter_by(name="John").count() == 1
 
 
-def test_edit(client):
-    assert client.post(url_for('role.edit', id=1)).status_code == HTTPStatus.NOT_FOUND
+def test_edit(client, db_session):
+    role = factories.RoleFactory(name="Manager")
+    client.post(url_for("role.edit", id=role.id), data={"name": "Admin"})
+
+    db_session.flush(role)
+    assert role.name == "Admin"
```

Comment:
@vryazanov I don't understand this test, it doesn't actually edit anything, or? It just posts the role and then flushes the DB? :D

Reply:
@vryazanov I would post (create) an original ``Role``, then post (edit) the same ``Role`` with some modifications, then simply ``query`` (using ``Role.query``) by id and see that changes are done. nothing with accessing the ``db_session`` directly :D

Reply:
@amihaiemil It creates `Role` using factories, then send new name for this role. and then just refresh `role` from DB and checks that there is correct value

Reply:
@amihaiemil in this case
`db_session.flush(role)` and `Role.query.get(role.id)` do the same thing

Reply:
@vryazanov and is the ``fush`` necessary? I mean, does the actual ``edit`` procedure also flush the DB? In my opinion the test should never have to do additional steps (like refreshing) to see if the code under test did something (but that's just my understanding, maybe Im wrong.

Reply:
@amihaiemil anyway I'll change to `Role.query` to be more understandable. I know that `.flush` always confuses people :)